File: metrics/timing_metrics.go
package metrics

import (
	"apt_cacher_go/utils/atomics"
	"time"
)

type timingMetrics struct {
	StartTime atomics.Time `json:"start_time"`
}

func NewTimingMetrics() timingMetrics {
	return timingMetrics{
		StartTime: atomics.NewAtomicTime(time.Now()),
	}
}

File: webserver/dashboard/frontend/src/lib/api/objects/metrics/timing-metrics.ts
import { readPropOrDefault } from "$lib/utils/values";
import { apiGet, type FetchFn } from "../../api-object";

export class TimingMetrics {
    readonly startTime: Date;

    constructor(json: Record<string, unknown>) {
        this.startTime = new Date(readPropOrDefault("start_time", json, ""));
    }
}

export async function getTimingMetrics(fetchFn: FetchFn = fetch): Promise<TimingMetrics> {
    return apiGet("/metrics/timing", TimingMetrics, fetchFn);
}
